include/boost/geometry/strategies/geographic/distance_cross_track.hpp

```diff
@@ -431,21 +486,59 @@ private :
 #endif
     }
 
-    template <typename Units, typename CT>
-    result_distance_point_segment<CT>
-    static inline apply(CT const& lo1, CT const& la1, //p1
-                        CT const& lo2, CT const& la2, //p2
-                        CT const& lo3, CT const& la3, //query point p3
-                        Spheroid const& spheroid)
+    
+    template <typename CT>
+    static inline auto non_iterative_case(CT const& lon1, CT const& lat1, //p1
+                                          CT const& lon2, CT const& lat2, //p2
+                                          CT const& distance)
+    {
+        result_type<CT> result;
+
+        set_result<EnableClosestPoint>::apply(distance, lon2, lat2, result);
+
+        return result;
+    }
+
+    template <typename CT>
+    static inline auto non_iterative_case(CT const& lon1, CT const& lat1, //p1
+                                          CT const& lon2, CT const& lat2, //p2
+                                          Spheroid const& spheroid)
     {
-        typedef typename FormulaPolicy::template inverse<CT, true, true, false, false, false>
-                inverse_dist_azimuth_type;
-        typedef typename FormulaPolicy::template inverse<CT, true, true, true, false, false>
-                inverse_dist_azimuth_reverse_type;
+        CT distance = geometry::strategy::distance::geographic
+            <
+                FormulaPolicy, 
+                Spheroid, 
+                CT
+            >::apply(lon1, lat1, lon2, lat2, spheroid);
+
+        return non_iterative_case(lon1, lat1, lon2, lat2, distance);
+    }
+
+protected:
+
+    template <typename CT>
+    static inline auto apply(CT const& lo1, CT const& la1, //p1
+                             CT const& lo2, CT const& la2, //p2
+                             CT const& lo3, CT const& la3, //query point p3
+                             Spheroid const& spheroid)
+    {
+        using inverse_dist_azimuth_type = typename FormulaPolicy::template 
+            inverse<CT, true, true, false, false, false>;
+
+        using inverse_dist_azimuth_reverse_type = typename FormulaPolicy::
+            template inverse<CT, true, true, true, false, false>;
```

Comment:
Should we pick one formatting here or is this ok?

Reply:
what about 
```
using inverse_dist_azimuth_type = 
            typename FormulaPolicy::template inverse<CT, true, true, false, false, false>;

using inverse_dist_azimuth_reverse_type = 
            typename FormulaPolicy::template inverse<CT, true, true, true, false, false>;
```
to be consistent with above?